packages/core/src/components/axes/zoom-bar.ts

```diff
@@ -417,10 +456,37 @@ export class ZoomBar extends Component {
 		}
 	}
 
+	// TODO - zoom bar skeleton could be improved in the future
+	renderSkeleton(container, startX, endX) {
+		// need to clear current zoom bar area
+		this.renderZoomBarArea(
+			container,
+			"path.zoom-graph-area-unselected",
+			[],
+			null
+		);
+		this.renderZoomBarArea(
+			container,
+			"path.zoom-graph-area",
+			[],
+			this.clipId
+		);
+		// remove brush listener
+		this.brush.on("start brush end", null);
+		// clear d3 brush
+		DOMUtils.appendOrSelect(
+			this.getContainerSVG(),
+			this.brushSelector
+		).html(null);
+		// re-render baseline
+		this.renderZoomBarBaseline(container, startX, endX);
+	}
```

Comment:
this is really unreadable code

Reply:
@theiliad 
What's your question ? 
When user change a chart from a normal state to the skeleton state, we need to remove the old d3 graphs and render the skeleton.
And since skeleton don't have axis labels, so the length of zoom bar will change. That's why I need to render baseline again with new x range.